src/vs/code/node/cli.ts

```diff
@@ -328,7 +328,25 @@ export async function main(argv: string[]): Promise<any> {
 			addArg(argv, '--no-sandbox'); // Electron 6 introduces a chrome-sandbox that requires root to run. This can fail. Disable sandbox via --no-sandbox
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (isMacintosh) {
+			let appArgs = ['-a', process.execPath];
+			if (args.wait) {
+				appArgs.push('--wait-apps');
+			}
+			const cliArgs = argv.slice(2);
+			const resolvedArgs = cliArgs.map(arg => {
```

Comment:
open doesn't seem to like relative paths in its arguments, hence this weird helper.

Reply:
I'm pretty sure this will break other features in unexpected ways... 😖

Reply:
Any thoughts on how to make this better ? I am not happy with the helper too, open for ideas.

Reply:
have updated to be strict only for file paths